js/newsletter.js

```diff
@@ -0,0 +1,148 @@
+jQuery(function() {
+
+	const verifyEmail = function(input) {
+		const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+		const $this = jQuery(input);
+
+		if (re.test($this.val())=== false) {
+			$this.addClass('error');
+			return false;
+		} 
+		$this.removeClass('error');
+		return true
+	}
+
+	const clearErrors = function(input) {
+		input.on('blur', function() {
+			verifyEmail(this);
+		});
+	}
+
+	const updateUserMeta = function() {
+		const url = "/wp-admin/admin-ajax.php?action=newsletter_subscribe";
+		jQuery.ajax({
+			url,
+			method: "POST",
+			data: {
+				subscribed: '1',
+			},
+			success: function(resp) {
+				newsletterThanks()
+			}
+		})
+	}
+
+	const newsletterFailure = function() {
+		const $newsletterForm = jQuery('.newsletter__form');
+		$newsletterForm.css('display', 'none');
+		const $failure = jQuery('.newsletter__failure');
+		$failure.css('display', 'block');
+	}
+
+	const newsletterError = function(error) {
+		const url = "/wp-admin/admin-ajax.php?action=newsletter_subscribe";
+		jQuery.ajax({
+			url,
+			method: "POST",
+			data: {
+				subscribed: '2',
+			},
+			success: function(resp) {
+				newsletterFailure()
+			}
+		})
+	}
+
+    // show sucess message
+    function newsletterThanks() {
+		const $newsletterForm = jQuery('.newsletter__form');
+		$newsletterForm.css('display', 'none');
+		const $success = jQuery('.newsletter__success');
+		$success.css('display', 'block');
+    }
+
+    // XHR subscribe; handle errors; display thanks message on success.
+    const handleSubmit = function(e) {
+		e.preventDefault();
+		const $this = jQuery(this);
+		const skipXHR = $this.attr('data-skip-xhr');
+		const $emailInput = jQuery('.newsletter__form input[name=email]');
+		validEmail = verifyEmail($emailInput);
+		clearErrors($emailInput);
+		const $privacyCheckbox = jQuery('#privacy');
+		const privacyCheck = $privacyCheckbox.prop('checked');
+		const $cpgError = jQuery('.newsletter__cpg__error');
+
+		if (!privacyCheck) {
+			$cpgError.addClass('newsletter__cpg__error--active');
+		}
+
+		if (!validEmail || !privacyCheck) {
+			return;
+		}
+
+        if (skipXHR) {
+			return true;
+        }
+
+		const email = $emailInput.val();
+        var params = 'email=' + encodeURIComponent(email) +
```

Comment:
For the profile submit you append country and language to the params, should it happen here as well?  doesn't look like those values are being sent.

Reply:
Good catch, @dnaluz! Pushed a fix for this. Also restricted the language list to match what is currently supported on their current newsletter sign up. Was getting an error if I tried to provide a language that wasn't listed here: https://activate.mozilla.community/newsletter